Update docs with verification-failure-behavior (#2962)
buildkite/docs: pages/agent/v3/signed_pipelines.md

This ensures that whenever those agents upload steps to Buildkite, they'll generate signatures using the private key you generated earlier. It also ensures that those agents verify the signatures of any steps they run, using the public key.

```ini
verification-failure-behavior=<warn>
```

> This setting determines the BuildKite agent’s response when it receives a job without a proper signature. It specifies how strictly the agent should enforce signature verification for incoming jobs. The agent will warn about the missing signature but will still proceed with executing the job. If not explicitly specified, the default behavior is `block`, which will prevent any job without a signature from running, ensuring a secure pipeline environment by default.

Reviewer: ```suggestion
This setting determines the Buildkite agent’s response when it receives a job without a proper signature. It specifies how strictly the agent should enforce signature verification for incoming jobs. The agent will warn about the missing signature but will still proceed with executing the job. If not explicitly specified, the default behavior is `block`, which will prevent any job without a signature from running, ensuring a secure pipeline environment by default.
```

> This setting determines the Buildkite agent’s response when it receives a job without a proper signature. It specifies how strictly the agent should enforce signature verification for incoming jobs. The agent will warn about the missing signature but will still proceed with executing the job. If not explicitly specified, the default behavior is `block`, which will prevent any job without a signature from running, ensuring a secure pipeline environment by default.

Reviewer: ```suggestion
This setting determines the Buildkite agent’s response when it receives a job without a proper signature. It specifies how strictly the agent should enforce signature verification for incoming jobs. The agent will warn about missing or invalid signatures, but will still proceed with executing the job. If not explicitly specified, the default behavior is `block`, which will prevent any job without a valid signature from running, ensuring a secure pipeline environment by default.
```

The usage string in the agent should be fixed: https://github.com/buildkite/agent/pull/2984

> This setting determines the Buildkite agent’s response when it receives a job without a proper signature. It specifies how strictly the agent should enforce signature verification for incoming jobs. The agent will warn about missing or invalid signatures, but will still proceed with executing the job. If not explicitly specified, the default behavior is `block`, which will prevent any job without a valid signature from running, ensuring a secure pipeline environment by default.

Reviewer 2: Just a few minor wording tweaks...

```suggestion
This setting determines the Buildkite agent's response when it receives a job without a proper signature, and specifies how strictly the agent should enforce signature verification for incoming jobs. The agent will warn about missing or invalid signatures, but will still proceed with executing the job. If not explicitly specified, the default behavior is `block`, which prevents any job without a valid signature from running, ensuring a secure pipeline environment by default.
```

We do this a lot in our docs, but avoid starting sentences with the word 'It...' because its meaning can be ambiguous (usually assumed to be the last/direct object in the previous sentence). In this case, 'it' could be interpreted as 'proper signature', whereas I'm sure 'This setting' is what 'It...' is meant to refer to.

> This setting determines the Buildkite agent’s response when it receives a job without a proper signature. It specifies how strictly the agent should enforce signature verification for incoming jobs. The agent will warn about missing or invalid signatures, but will still proceed with executing the job. If not explicitly specified, the default behavior is `block`, which will prevent any job without a valid signature from running, ensuring a secure pipeline environment by default.
> 
> 

Reviewer 2: Just a few minor wording tweaks...

```suggestion
This setting determines the Buildkite agent's response when it receives a job without a proper signature, and also specifies how strictly the agent should enforce signature verification for incoming jobs. The agent will warn about missing or invalid signatures, but will still proceed to execute the job. If not explicitly specified, the default behavior is `block`, which prevents any job without a valid signature from running, ensuring a secure pipeline environment by default.
```

We do this a lot in our docs, but avoid starting sentences with the word 'It...' because its meaning can be ambiguous (usually assumed to be the last/direct object in the previous sentence). In this case, 'it' could be interpreted as 'proper signature', whereas I'm sure 'This setting' is what 'It...' is meant to refer to.
